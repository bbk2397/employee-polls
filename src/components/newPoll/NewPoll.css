.poll-new-area,
.poll-new-title,
.poll-new-subtitle,
.poll-new-form,
.option-label,
.create-poll-button {
	margin: auto;
}

.poll-new-area {
	width: 70%;
}

.poll-new-title,
.poll-new-subtitle,
.option-label {
	text-align: center;
}

.poll-new-title {
	margin-bottom: 5px;
}

.poll-new-subtitle {
	color: grey;
}

.poll-new-form,
.option-group {
	display: grid;

	align-items: center;
	justify-content: center;
}

.poll-new-form {
	grid-template-columns: 90%;
	row-gap: 20px;

	padding-top: 30px;
}

.option-group {
	grid-template-columns: 100%;
	row-gap: 5px;
}

.option-label,
.create-poll-button {
	font-size: 20px;
}

.option-label {
	font-family: 'Courier New', monospace;
	font-weight: bold;	
}

.option-input {
	padding: 10px 5px;
	font-size: 16px;
}

.create-poll-button {
	background-color: white;
	border-color: white;
	cursor: pointer;
	border-color: #50C878;
	border-radius: 25px;
	width: 20%;
}

.create-poll-button:active {
	background-color: #50C878;
}

.create-poll-button:disabled {
	background-color: white;
	border-color: grey;
	cursor: default;
}
