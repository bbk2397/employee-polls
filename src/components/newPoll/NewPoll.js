import { connect } from "react-redux";
import { useState } from "react";
import { useNavigate } from "react-router-dom";
import { handleAddPoll } from "../../actions/polls";
import { setPathAfterLogin } from "../../actions/pathAfterLogin";
import "./NewPoll.css";

const NewPoll = ({ dispatch, loadingBar }) => {
	const [optionOneText, setOptionOneText] = useState('');
	const [optionTwoText, setOptionTwoText] = useState('');
	const navigate = useNavigate();

	const handleOptionOneChange = (e) => {
		e.preventDefault();
		setOptionOneText(e.target.value);
	}

	const handleOptionTwoChange = (e) => {
		e.preventDefault();
		setOptionTwoText(e.target.value);
	}

	const handleSubmit = (e) => {
		e.preventDefault();
		
		dispatch(handleAddPoll(
			optionOneText,
			optionTwoText,
			() => {
				dispatch(setPathAfterLogin('/'));
				navigate('/')
			}
		));
	}

	return (
		<div className="poll-new-area">
			<h2 className="poll-new-title">
				Would You Rather
			</h2>
			<h4 className="poll-new-subtitle">
				Create Your Own Poll
			</h4>
			<form
				onSubmit={handleSubmit}
				className="poll-new-form"
			>	
				<div className="option-group">
					<label
						htmlFor="option1"
						className="option-label"
					>
						First Option
					</label>
					<input
						type="text"
						name="option1"
						id="option1"
						className="option-input"
						data-testid="option-one-input"
						onChange={handleOptionOneChange}
					/>
				</div>
				<div className="option-group">
					<label
						htmlFor="option2"
						className="option-label"
					>
						Second Option
					</label>
					<input
						type="text"
						name="option2"
						id="option2"
						className="option-input"
						data-testid="option-two-input"
						onChange={handleOptionTwoChange}
					/>
				</div>
				<input
					type="submit"
					value="Create"
					className="create-poll-button"
					data-testid="submit-new-poll"
					disabled={
						optionOneText === "" ||
						optionTwoText === "" ||
						loadingBar.default > 0
					}
				/>
			</form>
		</div>
	);
}

const mapStateToProps = ({ loadingBar }) => ({
	loadingBar,
});

export default connect(mapStateToProps)(NewPoll);