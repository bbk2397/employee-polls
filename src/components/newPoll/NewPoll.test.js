import * as React from 'react';
import NewPoll from "./NewPoll";

import { Provider } from 'react-redux';
import store from "../../store";
import { MemoryRouter } from 'react-router';

import { render, fireEvent } from "@testing-library/react";

describe("NewPoll", () => {
	it("will have title and subtitle", () => {
		const component = render(
			<MemoryRouter>
			  <Provider store={store}>
		   		<NewPoll />
		 		</Provider>
	  	</MemoryRouter>
		);
		expect(component).toMatchSnapshot();
	});

	it("the button is disabled if none of the options are entered", () => {
		const component = render(
			<MemoryRouter>
			  <Provider store={store}>
		   		<NewPoll />
				</Provider>
			</MemoryRouter>
		);

		var submitButton = component.getByTestId('submit-new-poll');
		expect(submitButton.disabled).toBe(true);
	});

	it("the button is disabled if first option is not entered", () => {
		const component = render(
			<MemoryRouter>
			  <Provider store={store}>
		   		<NewPoll />
				</Provider>
			</MemoryRouter>
		);

		var submitButton = component.getByTestId('submit-new-poll');
		var optionTwoInput = component.getByTestId('option-two-input');
		fireEvent.change(optionTwoInput, { target: { value: 'B' } });
		expect(submitButton.disabled).toBe(true);
	});

	it("the button is disabled if second option is not entered", () => {
		const component = render(
			<MemoryRouter>
			  <Provider store={store}>
		   		<NewPoll />
				</Provider>
			</MemoryRouter>
		);

		var submitButton = component.getByTestId('submit-new-poll');
		var optionOneInput = component.getByTestId('option-one-input');
		fireEvent.change(optionOneInput, { target: { value: 'A' } });
		expect(submitButton.disabled).toBe(true);
	});

	it("the button is enabled if both options are entered", () => {
		const component = render(
			<MemoryRouter>
			  <Provider store={store}>
		   		<NewPoll />
				</Provider>
			</MemoryRouter>
		);

		var submitButton = component.getByTestId('submit-new-poll');
		var optionOneInput = component.getByTestId('option-one-input');
		fireEvent.change(optionOneInput, { target: { value: 'A' } });
		var optionTwoInput = component.getByTestId('option-two-input');
		fireEvent.change(optionTwoInput, { target: { value: 'B' } });
		expect(submitButton.disabled).not.toBe(true);
	});
});