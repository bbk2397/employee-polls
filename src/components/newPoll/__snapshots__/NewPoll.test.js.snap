// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPoll will have title and subtitle 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="poll-new-area"
      >
        <h2
          class="poll-new-title"
        >
          Would You Rather
        </h2>
        <h4
          class="poll-new-subtitle"
        >
          Create Your Own Poll
        </h4>
        <form
          class="poll-new-form"
        >
          <div
            class="option-group"
          >
            <label
              class="option-label"
              for="option1"
            >
              First Option
            </label>
            <input
              class="option-input"
              data-testid="option-one-input"
              id="option1"
              name="option1"
              type="text"
            />
          </div>
          <div
            class="option-group"
          >
            <label
              class="option-label"
              for="option2"
            >
              Second Option
            </label>
            <input
              class="option-input"
              data-testid="option-two-input"
              id="option2"
              name="option2"
              type="text"
            />
          </div>
          <input
            class="create-poll-button"
            data-testid="submit-new-poll"
            disabled=""
            type="submit"
            value="Create"
          />
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="poll-new-area"
    >
      <h2
        class="poll-new-title"
      >
        Would You Rather
      </h2>
      <h4
        class="poll-new-subtitle"
      >
        Create Your Own Poll
      </h4>
      <form
        class="poll-new-form"
      >
        <div
          class="option-group"
        >
          <label
            class="option-label"
            for="option1"
          >
            First Option
          </label>
          <input
            class="option-input"
            data-testid="option-one-input"
            id="option1"
            name="option1"
            type="text"
          />
        </div>
        <div
          class="option-group"
        >
          <label
            class="option-label"
            for="option2"
          >
            Second Option
          </label>
          <input
            class="option-input"
            data-testid="option-two-input"
            id="option2"
            name="option2"
            type="text"
          />
        </div>
        <input
          class="create-poll-button"
          data-testid="submit-new-poll"
          disabled=""
          type="submit"
          value="Create"
        />
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
